magicctapipe/scripts/lst1_magic/lst1_magic_stereo_reco.py

```diff
@@ -128,88 +112,91 @@ def stereo_reconstruction(
+
+    # It sometimes happens that there are MAGIC events whose event and
+    # telescope IDs are duplicated, so here we exclude those events
+    event_data.drop_duplicates(
+        subset=["obs_id", "event_id", "tel_id"], keep=False, inplace=True
+    )
     event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
     event_data.sort_index(inplace=True)
 
     is_simulation = "true_energy" in event_data.columns
+    logger.info(f"\nIs simulation: {is_simulation}")
 
     subarray = SubarrayDescription.from_hdf(input_file)
     tel_positions = subarray.positions
 
     logger.info("\nSubarray configuration:")
     for tel_id in subarray.tel.keys():
         logger.info(
-            f"\tTelescope {tel_id}: {subarray.tel[tel_id].name}, "
-            f"position = {tel_positions[tel_id]}"
+            f"\t{TEL_NAMES[tel_id]}: {subarray.tel[tel_id].name}, "
+            f"position = {tel_positions[tel_id].round(2)}"
         )
 
-    # Apply the event cuts:
-    if magic_only_analysis:
-        event_data.query("tel_id > 1", inplace=True)
+    # Apply the event cuts
+    quality_cuts = config_stereo["quality_cuts"]
 
-    config_stereo = config["stereo_reco"]
+    logger.info(
+        f"\nMAGIC-only analysis: {magic_only_analysis}"
+        f"\nQuality cuts: {quality_cuts}"
+    )
 
-    logger.info("\nApplying the quality cuts:")
-    logger.info(config_stereo["quality_cuts"])
+    if magic_only_analysis:
+        event_data.query("tel_id > 1", inplace=True)
 
-    event_data.query(config_stereo["quality_cuts"], inplace=True)
-    event_data["multiplicity"] = event_data.groupby(["obs_id", "event_id"]).size()
-    event_data.query("multiplicity > 1", inplace=True)
+    event_data = get_stereo_events(event_data, quality_cuts)
 
-    combo_types = check_tel_combination(event_data)
-    event_data[combo_types.columns[0]] = combo_types
+    # Check the angular distance of the LST-1 and MAGIC pointing directions
+    tel_ids = np.unique(event_data.index.get_level_values("tel_id")).tolist()
 
-    # Check the angular distance of the LST1 and MAGIC pointing directions:
-    if (not is_simulation) and (not magic_only_analysis):
+    if (not is_simulation) and (tel_ids != [2, 3]):
 
         logger.info(
-            "\nChecking the angular distance of the "
+            "\nCalculating the angular distance of the "
             "LST-1 and MAGIC pointing directions..."
         )
 
         theta = calculate_pointing_separation(event_data)
+        theta_max = theta.to(u.arcmin).max()
+        theta_uplim = u.Quantity(config_stereo["theta_uplim"])
 
-        if len(theta) == 0:
-            logger.info("--> No LST-1 + MAGIC events are found.")
+        if theta_max > theta_uplim:
+            logger.info(
+                f"--> The maximum angular distance is {theta_max.round(3)}, "
+                f"which is larger than the limit {theta_uplim}. Exiting..."
+            )
+            sys.exit()
```

Comment:
maybe instead of exiting simply the events with this large offset can be filtered out. running over subruns it does not matter much, because we loose only 10s or so, but this check is problematic when running over whole runs. Removing the events should just clip the (sub)-run so I do not think there should be any problems with the effective time calculations later

Reply:
OK, I changed the script to do so. Thanks for your comment.